src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp

```diff
@@ -238,7 +259,7 @@ bool WaypointMovementGenerator<Creature>::DoUpdate(Creature* creature, uint32 di
         {
             // xinef: code to detect pre-empetively if we should start movement to next waypoint
             // xinef: do not start pre-empetive movement if current node has delay or we are ending waypoint movement
-            bool finished = creature->movespline->Finalized();
+            bool finished = creature->movespline->currentPathIdx() < _transitionPointId;
             if (!finished && !i_path->at(i_currentNode)->delay && ((i_currentNode != i_path->size() - 1) || repeating))
```

Comment:
- !finished
+ finished

Reply:
In doing so, the creature does not spin in circles

Reply:
have fun with your crashes by accessing splines without checking if they are actually in progress

Reply:
https://github.com/TrinityCore/TrinityCore/pull/23864 I got it from here

Reply:
![image](https://user-images.githubusercontent.com/18535853/77415203-12ee9300-6dfd-11ea-879b-3fc61e29c78b.png)

Reply:
and you dont see the code is different? just take and ok? lol

Reply:
Btw, you don't want that code. Really. It's far from optimal and before adding more hacky stuff, just let it be.